lib/options-handler.js

```diff
@@ -280,11 +280,11 @@ export class ClientOptionsHandler {
         const copiedCompilers = JSON.parse(JSON.stringify(compilers));
         let semverGroups = {};
         // Reset the supportsExecute flag in case critical compilers change
-        _.each(this.options.languages, (lang) => {
-            lang.supportsExecute = false;
+        Object.keys(this.options.languages).map(key => {
+            this.options.languages[key].supportsExecute = false;
         });
 
-        _.each(copiedCompilers, (compiler, compilersKey) => {
+        for (const [compilersKey, compiler] of copiedCompilers.entries()) {
```

Comment:
Yeah, I'm not worried about the array destructuring, but you're calling .entries() on copiedCompilers itself, which is just plain old JSON data.

Reply:
I too found it weird but I referred to [this issue](https://stackoverflow.com/questions/34348937/access-to-es6-array-element-index-inside-for-of-loop#answer-34349073) to see how we can access `value` and `index` both using `for of` on the `copiedCompilers` array. This was also auto fixed by the lint rule.

The result was the same though on before and after data. Any alternate suggestions?

Reply:
Yes, again, I'm not worried about the concept of iterating over key+value, but it's this part which does not seem right `of copiedCompilers.entries()` as copiedCompilers is just JSON data and does not have a .entries() method

Reply:
I think it should be ok because as per the return type of the `JSON.parse` in the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse#return_value), it returns a valid `Array` or `Object`.

Since we are using `JSON.parse` on the `compilers`, I feel it should produce an iterable having a valid `Object.entries` function.

Thoughts? I might be missing something here.

Reply:
Apologies, stepped through with the debugger to inspect the data myself and it seems to be fine.

`copiedCompilers` is an array, which will have the .entires() method.